### PR TITLE
refactor(typings): es2015 module export + use updated AbstractLevel typings

### PR DESCRIPTION
**Pull request by MeirionHughes**
> ~req: https://github.com/Level/leveldown/pull/392~
> ~dep: https://github.com/Level/abstract-leveldown/pull/129~

### CODE REVIEW

File: leveldown.d.ts [hunk 1]
```diff
@@ -1,79 +1,74 @@
...
-declare function leveldown<
-  TKey=any,
-  TValue=any
-  >(location: string)
-  : leveldown.LevelDown<TKey, TValue>;
 
-export = leveldown;
\ No newline at end of file
+declare const LevelDown: LevelDownConstructor
+export default LevelDown;
```
Comment (zixia) [inline leveldown.d.ts]: I'd like to suggest also `export { LevelDown }`, the reason is as same as the other reviews.

Comment (MeirionHughes) [reply leveldown.d.ts]: sure, I'll update the pr's in awhile.

Comment (MeirionHughes) [reply leveldown.d.ts]: @zixia why do you want `export { LevelDown }` too? 

the `LevelDown` interface is already exported further up and can be imported with 

```ts
import { LevelDown } from './leveldown';
```

**Comment by MeirionHughes** [comment]
@ralphtheninja good to go on this; patch is fine.

**Comment by ralphtheninja** [comment]
@MeirionHughes We will release a new major since we removed support for 0.12

**Pull request merged**